From 7d90f7603af6b59e7144cef6617a1e9dd42161bd Mon Sep 17 00:00:00 2001
From: Jes Sorensen <jsorensen@fb.com>
Date: Mon, 18 May 2020 20:19:53 -0400
Subject: [PATCH 80/89] Include count for \0 character when using strncpy to
 implement strdup.
Git-commit: 7d90f7603af6b59e7144cef6617a1e9dd42161bd
Patch-mainline: mdadm-4.1+
References: jsc#SLE-13700

We have to include the \0 character in the length when copying a
string with strncpy() for which length was found with strlen().
Otherwise the destination will not get null terminated - except that
we explicitly zeroed it out earlier.

This quiets down the compiler's warnings.

Signed-off-by: Jes Sorensen <jsorensen@fb.com>
Signed-off-by: Coly Li <colyli@suse.de>
---
 dlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlink.c b/dlink.c
index 3efa94b..69aa7aa 100644
--- a/dlink.c
+++ b/dlink.c
@@ -63,7 +63,7 @@ char *dl_strndup(char *s, int l)
     if (s == NULL)
 	return NULL;
     n = dl_newv(char, l+1);
-    strncpy(n, s, l);
+    strncpy(n, s, l+1);
     n[l] = 0;
     return n;
 }
-- 
2.26.2

